app/knight/knight.py

```diff
@@ -0,0 +1,73 @@
+from app.armour.armour import Armour
+from app.potion.potion import Potion
+from app.weapon.weapon import Weapon
+
+
+class Knight:
+    def __init__(self, name: str, power: int,
+                 hp: int, weapon: dict,
+                 armour: list,
+                 potion: dict):
+        self.name = name
+        self.power = power
+        self.hp = hp
+        self.armour = Armour.get_total_armour(armour)
+        self.weapon = Weapon.get_weapon_cls(weapon)
+        self.potion = Potion.get_potion_cls(potion)
+
+    def __repr__(self):
+        return self.name
+
+    # create Knight instance
+    @classmethod
+    def get_knight_cls(cls, knight: dict) -> "Knight":
+        return cls(
+            name=knight["name"],
+            power=knight["power"],
+            hp=knight["hp"],
+            armour=knight["armour"],
+            weapon=knight["weapon"],
+            potion=knight["potion"]
+        )
+
+    # knight's statistics before the battle
+    def get_knight_stats(self):
+        setattr(self,
+                "total_hp",
+                self.hp + self.potion.hp)
+        setattr(self,
```

Comment:
Because I wanted to create new variables (total_hp, total_power, total_armour) which would contain corresponding parameters. There was two ways - directly (self.total_hp = ) and setattr. I suppoused that setattr would be better.

Reply:
which solution is preferred?

Reply:
directly way is preferred

Reply:
ok, thank you)